security/pkg/stsservice/tokenmanager/google/tokenexchangeplugin.go

```diff
@@ -493,12 +494,35 @@ func (p *Plugin) DumpPluginStatus() ([]byte, error) {
 	return statusJSON, err
 }
 
+// GetMetadata returns the metadata headers related to the token
+func (p *Plugin) GetMetadata(forCA bool, xdsAuthProvider, token string) (map[string]string, error) {
+	if token == "" {
+		return nil, fmt.Errorf("empty token in plugin GetMetadata")
+	}
+	gcpProjectNumber := p.GetGcpProjectNumber()
+	if !forCA && xdsAuthProvider == GCPAuthProvider && len(gcpProjectNumber) > 0 {
+		return map[string]string{
+			"authorization":       "Bearer " + token,
+			"X-Goog-User-Project": gcpProjectNumber,
+			"project":             "projects/" + gcpProjectNumber,
```

Comment:
What is the "project" header used for?

Reply:
This "project" header and the "X-Goog-User-Project" header are requested by @qfel to add gcp project number.

Reply:
You approved the change to drop the project header on the server side, so it should be fine to drop it here too now. BTW it can be *either* number or ID (for both headers).

Reply:
@qfel Drop both headers (the"project" header and the "X-Goog-User-Project" header ) or drop only the "project" header?

Reply:
Only "project".

Reply:
BTW for consistency you likely want to use x-goog-user-project (lowercase). HTTP headers are supposed to be case-insensitive, but HTTP2 (gRPC) seems to canonicalize to lowercase and other headers here (like authorization) are lowercase.

Reply:
Done: removed the project header and use x-goog-user-project (lowercase).